File: Scripts/DungeonGeneration/DungeonCreation/Room.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Room : MonoBehaviour
{
    public enum RoomSize
    {
        small,
        medium,
        large
    }
    public enum RoomType
    {
        spawn,
        room,
        boss,
        hall
    }
    public RoomSize roomSize;
    public RoomType roomType;
    public List<GameObject> doorWays;
}

File: Scripts/DungeonGeneration/DungeonCreation/room.py
from dataclasses import dataclass
from enum import Enum
from typing import List

from Scripts.DungeonGeneration.DungeonCreation.door_way import DoorWay


@dataclass
class Room:

    class RoomSize(Enum):
        SMALL = 1
        MEDIUM = 2
        LARGE = 3

    class RoomType(Enum):
        SPAWN = 1
        ROOM = 2
        BOSS = 3
        HALL = 4

    room_size: RoomSize
    room_type: RoomType
    door_ways: List[DoorWay]
